### PR TITLE
CR-1258 | improving logging in rest exception handler

### PR DESCRIPTION
**Pull request by akritisinhaa**
Updated the logging within the RestExceptionHandler

### CODE REVIEW

**Review by rpcatling — CHANGES_REQUESTED** [review]
These are not the changes I expected. I thought we discussed removing the stack trace from anything that was not level ERROR.

**Comment by akritisinhaa** [comment]
@rpcatling Changes made as requested. Please check this is what you expected. Also I have left the changes I made before because I felt there should be better logging message than just saying "Handling CTPException"

File: src/main/java/uk/gov/ons/ctp/common/error/RestExceptionHandler.java [hunk 1]
```diff
@@ -59,17 +59,25 @@
 
     switch (status) {
       case NOT_FOUND:
+        log.with("fault", exception.getFault())
+            .with("message", exception.getMessage())
+            .warn("Handling CTPException - The resource looking for is not found");
```
Comment (rpcatling) [inline src/main/java/uk/gov/ons/ctp/common/error/RestExceptionHandler.java:64]: English could be better, suggest:
```
Handling CTPException - The resource was not found
```

File: src/main/java/uk/gov/ons/ctp/common/error/RestExceptionHandler.java [hunk 2]
```diff
@@ -59,17 +59,25 @@
...
       case BAD_REQUEST:
+        log.with("fault", exception.getFault())
+            .with("message", exception.getMessage())
+            .warn("Handling CTPException - The request made is a bad request");
+        break;
       case ACCEPTED:
         log.with("fault", exception.getFault())
             .with("message", exception.getMessage())
-            .warn("Handling CTPException", exception);
+            .warn("Handling CTPException - The request is accepted  but unable to process");
```
Comment (rpcatling) [inline src/main/java/uk/gov/ons/ctp/common/error/RestExceptionHandler.java:74]: picky, but remove extra space before 'but'

File: src/main/java/uk/gov/ons/ctp/common/error/RestExceptionHandler.java [hunk 3]
```diff
@@ -59,17 +59,25 @@
...
         break;
 
       default:
         log.with("fault", exception.getFault())
             .with("message", exception.getMessage())
-            .error("Handling CTPException", exception);
+            .error("Handling CTPException - System error has taken place", exception);
```
Comment (rpcatling) [inline src/main/java/uk/gov/ons/ctp/common/error/RestExceptionHandler.java:80]: The words 'has taken place' are unnecessary.

**Review by rpcatling — CHANGES_REQUESTED** [review]
Made some minor comments on the wording.
In addition can we please look at all the Logging error() method calls and ensure a stack trace is generated with the causing exception. 

Note that godaddy logging also adds extra form of the error method that allows the exception to be the first parameter. 
```
 void error(Throwable t, String format, Object... args);
```
This is helpful if we want vararg parameters at the end.

**Review by rpcatling — APPROVED** [review]

File: src/main/java/uk/gov/ons/ctp/common/error/RestExceptionHandler.java [hunk 4]
```diff
@@ -59,17 +59,25 @@
 
     switch (status) {
       case NOT_FOUND:
+        log.with("fault", exception.getFault())
+            .with("message", exception.getMessage())
+            .warn("Handling CTPException - The resource was not found");
```
Comment (devenrawat) [inline src/main/java/uk/gov/ons/ctp/common/error/RestExceptionHandler.java:64]: Message could be "Resource not found"

**Comment by akritisinhaa** [comment]
LGTM

**Review by devenrawat — COMMENTED** [review]
one minor

File: src/main/java/uk/gov/ons/ctp/common/error/RestExceptionHandler.java (hunk 4 above)
Comment (akritisinhaa) [reply src/main/java/uk/gov/ons/ctp/common/error/RestExceptionHandler.java]: done

**Review by devenrawat — APPROVED** [review]

**Pull request merged**